agent/agent.go

```diff
@@ -703,6 +736,96 @@ func (a *agent) setLifecycle(state codersdk.WorkspaceAgentLifecycle) {
 	}
 }
 
+// reportConnectionsLoop reports connections to the agent for auditing.
+func (a *agent) reportConnectionsLoop(ctx context.Context, aAPI proto.DRPCAgentClient24) error {
+	for {
+		select {
+		case <-a.reportConnectionsUpdate:
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+
+		for {
+			a.reportConnectionsMu.Lock()
+			if len(a.reportConnections) == 0 {
+				a.reportConnectionsMu.Unlock()
+				break
```

Comment:
Do we need a label here for clarity? `break` will always break the innermost loop but I always have trouble remembering that personally.

Reply:
I only use labels when I need to. I don't think we have examples in our code for labels breaking the inner loop. Do we?